### ISSUE TITLE
Keyboard input occurs in the wrong input field when selecting a value or the value is read-only

### ISSUE THREAD

**Issue by ErikSteiner** [keepassxreboot/keepassxc#6277]
## Overview
Keyboard input is made in the wrong input field or the selected value cannot be overwritten by keyboard input

## Steps to Reproduce
1. Open Database
2. Use search bar to get an entry (say GitHub)
3. Do a right click on that entry 
4. Click edit entry
5. You can see the cursor is at the end of the value in field "Title". Select the mail address that is written in the "username" field, because you want to change it.
6. You expect the highlighted mail address to be overwritten by the keyboard input. However, the keyboard input suddenly occurs in the field above "Title".

## Expected Behavior
1. From the example above, I would expect to see the mail address in field "username" changes. The moment I press the first key, the selected area is overwritten.

## Actual Behavior
1. Even if the mail address is selected, the cursor is still in the field above it. Therefore, keyboard input causes the input to appear at the end of the field above and does not replace the selected text. Only after the mail address has been completely marked with the cursor a second time and the keyboard entry is made, the selected text will be replaced.

## Context
You can see the same behavior with the search bar. I enter the name of an entry. I find it. I click on the entry in the search result. Then I use the mouse to highlight the search term in the search box. The highlighted search term is immune to my keyboard input. It is not overwritten. The same behavior can be observed in the "Edit entry" for password generation. In the respective entry I click on the cube symbol, for the generation of a password. I select the value for the length of the password, because I want to replace it with an input on the numpad. The input of the numpad is done in the password field above. The selected value is not changed.

[NOTE]: # ( Paste debug info from Help → About here )

> KeePassXC - Version 2.6.4
> Revision: 34a78f0
> 
> Qt 5.15.2
> Diagnosemodus ist deaktiviert.
> 
> Betriebssystem: Manjaro Linux
> CPU-Architektur: x86_64
> Kernel: linux 5.10.19-1-MANJARO
> 
> Aktivierte Erweiterungen:
> - Auto-Type
> - Browser-Integration
> - SSH-Agent
> - KeeShare (bestätigtes und unbestätigtes Teilen)
> - YubiKey
> - Secret-Service-Integration
> 
> Kryptographische Bibliotheken:
> - libgcrypt 1.9.2-unknown

[NOTE]: # ( Pick choices based on your environment )
Operating System: Linux
Desktop Env: Gnome
Windowing System: X11

```
>> lsb_release -a
LSB Version:	n/a
Distributor ID:	ManjaroLinux
Description:	Manjaro Linux
Release:	20.2.1
Codename:	Nibia
>> uname -srm
Linux 5.10.19-1-MANJARO x86_64
```

**Comment by droidmonkey** [comment]
This is definitely not a bug with our application, but perhaps with Qt or the interaction of Qt and your window manager. Are you sure you are running X11? This has been reported before under Wayland, although not reproducible.

**Comment by ErikSteiner** [comment]
I am running X11. I only changed it to run X11 not as root.
Moreover I tried to change the session to Wayland and check if the problem will come up there, too. Yes, it does.

I don't know how meaningful it is: I started my Manjaro-Live-USB-Stick yesterday. Namely the one with the minimal installation. Even though the stick is not persistent, I did an update of all programs etc. in order to install KeePassCX. I created a sample database in KeePass and at least I could not find any of the errors described here. I don't know how conclusive this is, because theoretically I would have had to reboot after all the updates. Which makes no sense with a live stick. However, I'd also hate to reinstall everything only to find that the error is back.

**Comment by droidmonkey** [comment]
If you can narrow down the cause of the problem we can add it to our faq. It certainly isn't with our program though, as you demonstrated.

**Comment by ErikSteiner** [comment]
I added a bug report at QT, see [QTBUG-91894](https://bugreports.qt.io/browse/QTBUG-91894).

Today I came up with the solution. I wrote a bug report at QT and wanted to add a screenshot. I didn't have a program installed that could be used to make something unrecognizable.

So I installed Krita. Krita however didn't launch. I tried it via Terminal: `Segmentation fault (core dumped)` and so I came to this post here: [Krita gives segmentation fault after update](https://forum.manjaro.org/t/krita-gives-segmentation-fault-after-update/53058/2). I read "This problem is caused by Deepin Terminal"

I looked into my installed programs and searched for deepin. I found deepin-album, a photo management tool that I installed some time ago. It depends on `deepin-qt5integration`, `freeimage`, `gio-qt`, `udisks2-qt5`. I uninstalled it and launched KeePass. The issue was solved.

I also tested another Program "OpenRGB", where I had similar issues, too. These issues were solved after removing deepin-album and its dependencies.

To confirm it, I set up a virtual machine with Manjaro. I installed KeePass and everything was fine. Then I installed deepin-album and the same issues occurred that I described above.

I am looking forward to the comments on QT.

**Comment by droidmonkey** [comment]
I think its also time to tell deepin or manjaro package maintainers to fix their software or get it removed from the distro!!

**Comment by droidmonkey** [comment]
Filed bug with Deepin -- https://github.com/linuxdeepin/qt5integration/issues/41

**Comment by kegechen** [comment]
please let us know the version of dde-qt5integration and 

> he wrong input field or the selected value cannot be overwritten by keyboard input
> 
> ## Steps to Reproduce

This is actually caused by dde-qt5xcb-plugin(<5.0.24),  QGuiApplication::styleHints()->setFocusOnTouchRelease();
and the xcb plugin override styleHint return true(fixed >=5.0.24), [fixed here](urlhttps://github.com/linuxdeepin/qt5platform-plugins/commit/f9a66fd415252249358af5f6a0a51eb73aa88da2#diff-ffd0cd5de3f7923ca99eaa500b5585a49ea3f6ac79b78351375d49f41f7ee2f5)

**Comment by droidmonkey** [comment]
This may be true for this particular bug, but I also installed deepin software on Fedora 33 to test another bug and found that it was interfering with typing into text edit fields and general keyboard navigation. Also the same bug was present there and dde was not installed since this was on Wayland.
